### DISCUSSION TITLE
REPLACE INTO equivalent

### DISCUSSION THREAD

**Discussion by sazalihisham88** [mikro-orm/mikro-orm#3552] [Q&A]
Hi I am still new in this ORM field. I wonder if i can achieve REPLACE INTO query equivalent using EntityManager? The use case is I need to insert thousands of rows, and if unique_key column value already exists, it needs to be overwritten with the new values.
Thanks!

**Comment by B4nan** [comment]
You can use QB with onConflict clause (check knex docs about this, the API is the same), EM is not suitable for such operations, as it works with in memory state, if you want update queries, you need to first load the entity there.

**Comment by sazalihisham88** [comment]
@B4nan 

With query builder, is that mean i need to execute it row by row? Is it possible to do it in bulk to have better execution time? Please bear with my lack of knowledge.

**Reply by B4nan** [reply]
You can use batch inserts, that's actually SQL feature, nothing specific to entity manager.
